src/Service.Host/client/src/components/importBooks/tabs/OrderDetailsTab.js

```diff
@@ -61,49 +61,74 @@ function OrderDetailsTab({
         updateRow({ ...row, [propertyName]: newValue });
     };
 
+    const calcRemainingTotal = () => {
+        const orderDetailsTotal = orderDetails?.reduce((a, v) => a + v.orderValue, 0);
+
+        let remaining = `${totalInvoiceValue - orderDetailsTotal}`;
+
+        if (remaining) {
+            remaining = `${parseFloat(remaining).toFixed(2)}`;
+        }
+
+        return remaining || 0;
+    };
+
+    const calcRemainingDuty = () => {
+        const orderDetailsDutyTotal = orderDetails?.reduce((a, v) => a + v.dutyValue, 0);
+
+        let remaining = `${duty - orderDetailsDutyTotal}`;
+
+        if (remaining) {
+            remaining = `${parseFloat(remaining).toFixed(2)}`;
+        }
+
+        return remaining || 0;
+    };
+
+    const calcRemainingWeight = () => {
+        const orderDetailsWeightTotal = orderDetails?.reduce((a, v) => a + v.weight, 0);
+
+        let remaining = `${weight - orderDetailsWeightTotal}`;
+
+        if (remaining) {
+            remaining = `${parseFloat(remaining).toFixed(2)}`;
+        }
+
+        return remaining || 0;
+    };
```

Comment:
I could be wrong, but you could risk running into floating point precision errors here doing totals etc. in Javascript. I tend to either do sums in the back end with c# decimal objects, or us a number javascript library that'll take care of this stuff for you. decimal.js is already in the project if you do decide to go down the js route

Reply:
I think you're probably right with that, it seemed ok at a glance but I know js does some unpredictable things with numbers. I'm going to put this through as I've directed some folk to this screen as an option to view impbooks, but I'll definitely change it and try using decimal.js in my next pr. Don't think there's any immediate danger here as only Yvonne and Rhona have edit permission and both know it isn't complete yet, so nothing will be being saved